core/arch/arm/plat-rockchip/platform_rk3588.c

```diff
@@ -160,3 +191,197 @@ TEE_Result hw_get_random_bytes(void *buf, size_t blen)
 
 	return TEE_SUCCESS;
 }
+
+static TEE_Result tee_otp_read_secure(uint32_t *pValue, uint32_t index,
+				      uint32_t count)
+{
+	vaddr_t base = (vaddr_t)phys_to_virt(OTP_S_BASE, MEM_AREA_IO_SEC,
+					     OTP_S_SIZE);
+	uint32_t int_status = 0;
+	uint32_t i = 0;
+	uint32_t val = 0;
+
+	/* Setup read: index, count, command = READ */
+	uint32_t auto_ctrl_val = (index << ADDR_SHIFT) |
+				 (count << BURST_SHIFT) |
+				 CMD_READ;
+
+	if (!base)
+		panic("OTP_S base not mapped");
+
+	/* Check for invalid parameters or exceeding hardware burst limit */
+	if (!pValue || !count || count > BURST_SIZE)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	/* Clear any pending interrupts by reading & writing back INT_ST */
+	io_write32(base + OTP_S_INT_ST,
+		   io_read32(base + OTP_S_INT_ST));
+
+	/* Set read command */
+	io_write32(base + OTP_S_AUTO_CTRL, auto_ctrl_val);
+
+	/* Enable read */
+	io_write32(base + OTP_S_AUTO_EN, EN_ENABLE);
+
+	/* Wait for RD_DONE bit using IO_READ32_POLL_TIMEOUT */
+	if (IO_READ32_POLL_TIMEOUT(base + OTP_S_INT_ST,
+				   int_status,
+				   int_status & OTP_S_RD_DONE_BIT,
+				   OTP_POLL_PERIOD_US,
+				   OTP_POLL_TIMEOUT_US)) {
+		return TEE_ERROR_BUSY;
+	}
+
+	/* Clear the interrupt again */
+	io_write32(base + OTP_S_INT_ST,
+		   io_read32(base + OTP_S_INT_ST));
+
+	/* Read out the data */
+	for (i = 0; i < count; i++) {
+		val = io_read32(base + OTP_S_DOUT +
+				(i * sizeof(uint32_t)));
+		pValue[i] = val;
+	}
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result tee_otp_write_secure(const uint32_t *pValue, uint32_t index,
+				       uint32_t count)
+{
+	vaddr_t base = (vaddr_t)phys_to_virt(OTP_S_BASE, MEM_AREA_IO_SEC,
+					     OTP_S_SIZE);
+	uint32_t int_status = 0;
+
+	if (!base)
+		panic("OTP_S base not mapped");
+
+	/* Check for invalid parameters or exceeding hardware limits */
+	if (!pValue || !count || count > BURST_SIZE ||
+	    (index + count > MAX_INDEX))
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	/* Program OTP words */
+	for (uint32_t i = 0; i < count; i++) {
+		uint32_t old_val, new_val = 0;
+		uint32_t curr_idx = index + i;
+		TEE_Result res = 0;
```

Comment:
prefer a known `TEE_Result` value instead of `0`, e.g. `TEE_SUCCESS` or `TEE_ERROR_GENERIC`.